editions/tw5.com/tiddlers/widgets/CheckboxWidget.tid

```diff
@@ -2,9 +2,25 @@ caption: checkbox
 created: 20131024141900000
 modified: 20211009121239806
 tags: Widgets TriggeringWidgets
+colors: red orange yellow blue
 title: CheckboxWidget
 type: text/vnd.tiddlywiki
 
+\define checkActions() <$action-listops $field="colors" $subfilter="-red green"/>
```

Comment:
Having the macro definitions separated from the examples is awkward.

The wikitext-example-without-html macro creates a new parsing context, so it is possible to make the examples self-contained by including the necessary definitions within. Is that possible in this case?

Reply:
Yeah, it certainly was awkward having to write it this way, with the macro definitions at the top of the tiddler. I'm glad to learn about the new parsing context (did not know that before); I'll rewrite these examples using that fact.